src-electron/db/query-bitmap.js

```diff
@@ -68,6 +68,62 @@ WHERE (DATA_TYPE.NAME = ? OR DATA_TYPE.NAME = ?) AND DATA_TYPE.PACKAGE_REF IN ($
     .then(dbMapping.map.bitmap)
 }
 
+/**
+ * Select a bitmap matched by name and clusterId.
+ * @param {*} db
+ * @param {*} name
+ * @param {*} clusterId
+ * @param {*} packageIds
+ * @returns bitmap information or undefined
+ */
+async function selectBitmapByNameAndClusterId(db, name, clusterId, packageIds) {
+  let res = await dbApi
+    .dbAll(
+      db,
+      `
+    SELECT
+      BITMAP.BITMAP_ID,
+      DATA_TYPE.NAME AS NAME,
+      BITMAP.SIZE AS SIZE
+    FROM BITMAP
+    INNER JOIN DATA_TYPE ON BITMAP.BITMAP_ID = DATA_TYPE.DATA_TYPE_ID
+    WHERE (DATA_TYPE.NAME = ? OR DATA_TYPE.NAME = ?) AND DATA_TYPE.PACKAGE_REF IN (${dbApi.toInClause(
+      packageIds
+    )})`,
+      [name, name.toLowerCase()]
+    )
+    .then((rows) => rows.map(dbMapping.map.bitmap))
+
+  if (res && res.length == 1) {
+    return res[0]
+  } else {
+    return dbApi
+      .dbGet(
+        db,
+        `
+SELECT
+  BITMAP.BITMAP_ID,
+  DATA_TYPE.NAME AS NAME,
+  BITMAP.SIZE AS SIZE
+FROM BITMAP
+INNER JOIN
+  DATA_TYPE
+ON
+  BITMAP.BITMAP_ID = DATA_TYPE.DATA_TYPE_ID
+INNER JOIN
+  DATA_TYPE_CLUSTER
+ON
+  DATA_TYPE_CLUSTER.DATA_TYPE_REF = BITMAP.BITMAP_ID
+WHERE
+  (DATA_TYPE.NAME = ? OR DATA_TYPE.NAME = ?)
+  AND DATA_TYPE_CLUSTER.CLUSTER_REF = ?
```

Comment:
Would it perhaps make more sense to match on cluster code, not the temporary-for-the-database id?  That would allow templates to do checks with known cluster codes as needed....  Again, same for the other places where we are adding these sorts of queries.

Reply:
I prefer cluster_ref since it is unique. Cluster code can sometimes be mixed up with mfg codes which then becomes a separate check for Zigbee pro. This handles both matter and zigbee pro. I would not change this unless really necessary.

Reply:
I left a note in all the helpers which are using clusterId